Source/Atomic/Math/StringHash.cpp

```diff
@@ -43,20 +43,24 @@ const StringHash StringHash::ZERO;
 StringHash::StringHash(const char* str) :
     value_(Calculate(str))
 {
+#if ATOMIC_PROFILING
+    RegisterSignificantString(str, *this);
+#endif

```

Comment:
Ok, this is definitely an area to keep an eye on, and maybe for future improvement.  It registers every time a string is converted to a StringHash, ouch... I know this is the only way right now, hopefully this doesn't come up on the top of the profiler ;)

It it does, maybe we can introduce a separate ```ATOMIC_PROFILING_EVENTS``` for enabling event profiling...